### DISCUSSION TITLE
Interoperability with C++ code

### DISCUSSION THREAD

**Discussion by k-s-s** [nxrighthere/UnrealCLR#93] [Help] [answered]
Hi

My project I've done some work in C++. Is there any way to reference enums/ classes from the C++ project, other than doing a round trip by blueprint?
Or any guide how to make a wrapper class/ project that can be referenced.

Thanks

**Comment by nxrighthere** [comment, marked as answer]
Yes, it's possible but requires a good understanding of how interoperability in C# works in general. Wrapping and exposing C++ functions is just regular P/Invoke with all relevant implications related to binary compatibility of function signatures and data types, memory management, and so on.

In one or another way you will need glue code, but with blueprints, it's way easier and safer. With blueprint nodes generation it will be even better, see https://github.com/nxrighthere/UnrealCLR/issues/33.
